### ISSUE TITLE
Macro for typed queries doesn't compile if slf4j-log4j or logback is present as a project dependency

### ISSUE THREAD

**Issue by alexdupre** [slick/slick#1174]
Sample minimal project that exhibits the issue: http://www.alexdupre.com/tarballs/slick-bug.tar.gz

```
> compile
[info] Updating {file:/usr/home/alex/workspace/slick-bug/}slick-bug...
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] Compiling 1 Scala source to /usr/home/alex/workspace/slick-bug/target/scala-2.11/classes...
[error] /usr/home/alex/workspace/slick-bug/src/main/scala/DB.scala:16: exception during macro expansion: 
[error] com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'slick'
[error]         at com.typesafe.config.impl.SimpleConfig.findKey(SimpleConfig.java:124)
[error]         at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:147)
[error]         at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:159)
[error]         at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:164)
[error]         at com.typesafe.config.impl.SimpleConfig.getBoolean(SimpleConfig.java:174)
[error]         at slick.util.GlobalConfig$.<init>(GlobalConfig.scala:15)
[error]         at slick.util.GlobalConfig$.<clinit>(GlobalConfig.scala)
[error]         at slick.util.TreeDump$.<init>(TreeDump.scala:9)
[error]         at slick.util.TreeDump$.<clinit>(TreeDump.scala)
[error]         at slick.backend.DatabaseComponent$DatabaseDef$class.logAction(DatabaseComponent.scala:322)
[error]         at slick.jdbc.JdbcBackend$DatabaseDef.logAction(JdbcBackend.scala:38)
[error]         at slick.backend.DatabaseComponent$DatabaseDef$class.runInContext(DatabaseComponent.scala:140)
[error]         at slick.jdbc.JdbcBackend$DatabaseDef.runInContext(JdbcBackend.scala:38)
[error]         at slick.backend.DatabaseComponent$DatabaseDef$class.runInternal(DatabaseComponent.scala:75)
[error]         at slick.jdbc.JdbcBackend$DatabaseDef.runInternal(JdbcBackend.scala:38)
[error]         at slick.backend.DatabaseComponent$DatabaseDef$class.run(DatabaseComponent.scala:72)
[error]         at slick.jdbc.JdbcBackend$DatabaseDef.run(JdbcBackend.scala:38)
[error]         at slick.jdbc.ActionBasedSQLInterpolation$.tsqlImpl(StaticQuery.scala:179)
[error]   def test(): Future[Seq[Int]] = db.run(tsql"select id from test")
[error]                                         ^
[error] one error found
[error] (compile:compileIncremental) Compilation failed
[error] Total time: 1 s, completed 22-giu-2015 17.43.10
```

**Comment by lucamolteni** [comment]
+1

**Comment by solidsystem** [comment]
I encountered the same problem, it looks like this is solved in 3.1.0-M1. The GlobalConfig.scala there has defaults for missing configuration keys.

As a temporary workaround until 3.1 is released, I copied [GlobalConfig.scala](https://github.com/slick/slick/blob/5a605371e8feb7f7e4e1c0a47664db890bf4a778/slick/src/main/scala/slick/util/GlobalConfig.scala) from 3.1.0-M1 into my fork of 3.0.1 and compiled slick myself. It seems to work fine for me now.

**Comment by szeiger** [comment]
I have no idea how this would be triggered by slf4j or logback but support for resources on the compiler classpath is indeed a problem in various environments, including sbt. The workaround in 3.1 is to have defaults for everything hardcoded so it can still run without reference.conf. You may still run into problems getting your application.conf loaded though, but it's generally easier and nothing we can fix on our end anyway.
